### ISSUE TITLE
Alternative sources

### ISSUE THREAD

**Issue by pwntester** [wbthomason/packer.nvim#416]
Hi, not sure if this is currently possible and I was just unable to figure out how to do it. But I  would like to set alternative sources for a plugin. For example, try to load it locally and if it fails, try to load it from GitHub. Is that possible already? Thanks!

**Comment by wbthomason** [comment]
Oh, fallback sources? That's an interesting idea that we don't currently support without some hacking. You could maybe implement this yourself using a custom key handler (https://github.com/wbthomason/packer.nvim#extending-packer), but it isn't yet possible.

**Comment by akinsho** [comment]
@pwntester out of curiosity, is this for handling plugins that are local on some systems but remote on others using the same plugin spec? I ask because I've worked on something like this in my own config, not using an official extension though

**Comment by pwntester** [comment]
Thanks @akinsho Yep, the use case is exactly that, have my own plugins loaded locally on my dev machine, but when working remotely (eg: GitHub CodeSpaces) load them from github.

Is this handled [here](https://github.com/akinsho/dotfiles/blob/58b2979f51b2f3e467e14e940b6a4aa63d1868f3/.config/nvim/lua/as/plugins/init.lua#L64-L113)?

**Comment by akinsho** [comment]
Yep that's the one, it's currently quite specific to my needs and messy, but the general idea is to create a local spec and a remote one, alias the name of the local one, so they don't clash. It essentially checks if the local directory exists and if it does, it creates a local spec, otherwise just a remote one.  It also loads one or the other based on some condition by actually using a `cond` key. For example, I only load local versions of plugins I contribute to when I set an env var otherwise I use the remote but for personal plugins I created I always load the local version.

It doesn't handle some cases well, e.g. things like specifying a `cmd` since both local and remote will try to both create the same command which will clash, but for simple cases it works pretty well 🤷🏾. It seemed to niche a use case to bother trying to contribute back to packer in some improved tidy form, since I think only a small percentage of users are also plugin makers, probably.
